core/primitives-core/src/serialize.rs

```diff
@@ -212,3 +216,27 @@ pub mod option_u128_dec_format {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct DeserializedWithDebug<T>(pub T);
+
+impl<T> DeserializedWithDebug<T> {
+    fn _deserialize<'de, D>(deserializer: D) -> Result<T, D::Error>
+    where
+        T: Deserialize<'de>,
+        D: Deserializer<'de>,
+    {
+        if cfg!(debug_assertions) {
+            let mut unused_fields = Vec::new();
+            let de_result: Result<T, D::Error> = serde_ignored::deserialize(deserializer, |path| {
+                unused_fields.push(path.to_string());
+            });
+
+            debug!("missed fields while deserializing: {:?}", unused_fields);
```

Comment:
Probably `warn!`.

Also an issue here is that this does not include information about what we were deserialising. Ideally it would be: `warn!("{}: encountered unrecognised fields: {:?}", path.display(), unused_fields);`. Perhaps it would make sense to change `DeserializedWithDebug` to have two fields? E.g. `pub struct DeserializedWithDebug<T>(pub T, pub Vec<String>)` so that the caller can decide how to format the warning? Though then we would be repeating pretty much the same `warn!` code in multiple places. Now that I think about it perhaps what we actually want is `load_json_from_file` function which would read the file, deserialise, log the warning and return the object?